corehq/apps/ota/views.py

```diff
@@ -139,21 +139,31 @@ def claim(request, domain):
     as_user_obj = CouchUser.get_by_username(as_user) if as_user else None
     restore_user = get_restore_user(domain, request.couch_user, as_user_obj)
 
-    case_id = unquote(request.POST.get('case_id', ''))
-    if not case_id:
+    # This will be expecting space seperated values of all the cases to be claimed
+    case_ids = request.POST.getlist('case_id')
+
+    if not case_ids:
         return HttpResponse('A case_id is required', status=400)
 
     try:
-        if get_first_claim(domain, restore_user.user_id, case_id):
-            return HttpResponse(status=204)
+        case_ids_already_claimed = get_first_claims(domain, restore_user.user_id, case_ids)
```

Comment:
I think you could simplify this code by pushing more of the logic into the helper function.  For instance, instead of `get_first_claims`, the helper could be `get_unclaimed_cases`.